src/main/java/io/tiledb/spark/TileDBScan.java

```diff
@@ -0,0 +1,75 @@
+package io.tiledb.spark;
+
+import static org.apache.spark.metrics.TileDBMetricsSource.dataSourceReadSchemaTimerName;
+
+import java.net.URI;
+import java.util.Map;
+import org.apache.log4j.Logger;
+import org.apache.spark.TaskContext;
+import org.apache.spark.metrics.TileDBReadMetricsUpdater;
+import org.apache.spark.sql.connector.metric.CustomMetric;
+import org.apache.spark.sql.connector.read.Batch;
+import org.apache.spark.sql.connector.read.Scan;
+import org.apache.spark.sql.connector.read.streaming.ContinuousStream;
+import org.apache.spark.sql.connector.read.streaming.MicroBatchStream;
+import org.apache.spark.sql.sources.Filter;
+import org.apache.spark.sql.types.StructType;
+
+public class TileDBScan implements Scan {
+  private final TileDBReadSchema tileDBReadSchema;
+  private final Map<String, String> properties;
+  private final TileDBDataSourceOptions options;
+  private final Filter[] pushedFilters;
+  private final URI uri;
+
+  private final TileDBReadMetricsUpdater metricsUpdater;
+  static Logger log = Logger.getLogger(TileDBScan.class.getName());
+
+  public TileDBScan(
+      TileDBReadSchema tileDBReadSchema,
+      Map<String, String> properties,
```

Comment:
What is the difference between properties and options? (other than `DataSourceOptions` is case-insensitive..?)

Reply:
None, good catch!  I removed the carrying of properties in the code, they are not necessary. We use the options throughout.